extensions/interactions/PencilCodeEditor/PencilCodeEditor.html

```diff
@@ -38,3 +37,19 @@
 <script type="text/ng-template" id="shortResponse/PencilCodeEditor">
   <pre><[answerCode | truncateAtFirstLine]></pre>
 </script>
+
+<script type="text/ng-template" id="modals/pencilCodeResetConfirmation">    
+  <div class="modal-header">
+    <h4 class="modal-title">
+      <span translate="I18N_INTERACTIONS_PENCILCODEEDITOR_CONFIRMATION_REQUIRED"></span>
+    </h4>
+  </div>
+  <div class="modal-body">
+    <span translate="I18N_INTERACTIONS_PENCILCODEEDITOR_ARE_YOU_SURE_YOU_WANT_TO_RESET_YOUR_CODE?">
+    </span>
+  </div>
+  <div class="modal-footer">
+    <button type="button" ng-click="cancel()" class="btn btn-default" data-dismiss="modal" translate="I18N_INTERACTIONS_PENCILCODEEDITOR_CANCEL"></button>
```

Comment:
data-dismiss closes the modal when it is clicked. It's part of bootstrap.
But I realize that we don't need it as we are handling click with ng-click. I'll remove it now.